### PR TITLE
DRY wait_boot/first_boot

### PR DESCRIPTION
**Pull request by okurz**
* Merge 'first_boot' and 'wait_boot' as much as possible
* wait_boot: Simplify by extracting methods
* first_boot: Remove obsolete temporary workaround for machine "hornet"
* Remove redundant module "first_boot" from "boot_to_snapshot" schedule

Verification runs:
* [opensuse-Tumbleweed-DVD-x86_64-extra_tests_on_kde_okurz_only_multi-users-dm@64bit](https://openqa.opensuse.org/tests/1099140)
* [gnome-gdm](https://openqa.opensuse.org/tests/1099076)
* [cryptlvm-uefi](https://openqa.opensuse.org/tests/1099077)
* [sle-15-SP2-create_hdd_sles4sap_textmode](https://openqa.suse.de/tests/3644110)
* [sle-12-SP5-SAP-DVD-x86_64-create_hdd_sles4sap_textmode](https://openqa.suse.de/tests/3644087)
* [sle-12-SP5-Server-DVD-x86_64-autologin_yast@64bit](https://openqa.suse.de/tests/3644089)
* [sle-12-SP5-Server-DVD-ppc64le-cryptlvm@ppc64le](https://openqa.suse.de/tests/3644091)
* [sle-12-SP5-Server-DVD-aarch64-cryptlvm@aarch64](https://openqa.suse.de/tests/3644092)
* [sle-12-SP5-Server-DVD-s390x-xfs@s390x-kvm-sle12](https://openqa.suse.de/tests/3644093)
* [sle-12-SP5-Server-DVD-s390x-xfs@s390x-zVM-vswitch-l2](https://openqa.suse.de/tests/3644094)
* [sle-12-SP4-Desktop-DVD-Updates-x86_64-qam-gnome@64bit](https://openqa.suse.de/tests/3644095)
* [sle-12-SP4-Server-DVD-Updates-x86_64-qam-textmode@64bit](https://openqa.suse.de/tests/3644096)
* [sle-15-Server-DVD-Updates-x86_64-mru-install-minimal-with-addons@64bit](https://openqa.suse.de/tests/3644097)
* [sle-15-Server-DVD-Updates-x86_64-qam-gnome@64bit](https://openqa.suse.de/tests/3644098)
* [opensuse-Tumbleweed-DVD-x86_64-boot_to_snapshot@64bit](https://openqa.opensuse.org/tests/1099074)
* [opensuse-Tumbleweed-DVD-x86_64-gnome@64bit](https://openqa.opensuse.org/tests/1098513)
* [opensuse-Tumbleweed-DVD-x86_64-kde@64bit](https://openqa.opensuse.org/tests/1098471)
* [opensuse-Tumbleweed-DVD-x86_64-minimalx@64bit](https://openqa.opensuse.org/tests/1098472)
* [opensuse-Tumbleweed-DVD-x86_64-textmode@64bit](https://openqa.opensuse.org/tests/1098473)
* [opensuse-Tumbleweed-DVD-x86_64-upgrade_Leap_15.1_gnome@64bit](https://openqa.opensuse.org/tests/1099143)
* [opensuse-Tumbleweed-DVD-x86_64-extra_tests_on_kde@64bit](https://openqa.opensuse.org/tests/1099141)
* [opensuse-Tumbleweed-DVD-x86_64-extra_tests_on_gnome@64bit](https://openqa.opensuse.org/tests/1099142)
* [opensuse-Tumbleweed-DVD-x86_64-extra_tests_in_textmode_okurz@64bit](https://openqa.opensuse.org/tests/1099075)

Related progress issue: https://progress.opensuse.org/issues/60401

### CODE REVIEW

**Comment by ldevulder** [comment]
Running verification run:
- [SLE-12-SP5-sles4sap_scc_gnome_sap_inst_wizard@64bit-ipmi](https://openqa.suse.de/tests/3644675)

Verification runs OK:
- [SLE-15-SP2-ppc64le-sles4sap_scc_gnome_saptune_v2_note_1410736](https://openqa.suse.de/tests/3644674)
- [SLE-15-x86_64-ppc64le-sles4sap_scc_gnome_saptune_v2_note_1410736](https://openqa.suse.de/tests/3644673)

Verification runs that failed:
- [SLE-12-SP5-DVD-x86_64-create_hdd_sles4sap_gnome](https://openqa.suse.de/tests/3644671)
- [SLE-15-SP2-DVD-x86_64-create_hdd_sles4sap_gnome](https://openqa.suse.de/tests/3644672)

I have an issue with `first_boot`: https://openqa.suse.de/tests/3644672#step/first_boot/5.
"Password" screen should appears but it doesn't, seems like the username is not entered.

**Comment by okurz** [comment]
@ldevulder thanks for testing. I added another commit to also remove some duplication in "handle_login" which helps:

Created job #1099358: opensuse-Tumbleweed-DVD-x86_64-extra_tests_on_kde@64bit -> https://openqa.opensuse.org/t1099358 -> failed in logout of multi_users_dm, retriggered as https://openqa.opensuse.org/tests/1099406 all good

```
for i in https://openqa.opensuse.org/tests/1099076 https://openqa.suse.de/tests/3644088 https://openqa.suse.de/tests/3644086 https://openqa.suse.de/tests/3644083 https://openqa.suse.de/tests/3644084 https://openqa.suse.de/tests/3644098 ; do openqa-clone-job --parental-inheritance --skip-chained-deps --within-instance $i BUILD= _GROUP=0 CASEDIR=https://github.com/okurz/os-autoinst-distri-opensuse.git#enhance/wait_boot2; done
```

* [opensuse-Tumbleweed-DVD-x86_64-gnome-gdm@64bit](https://openqa.opensuse.org/t1099343)
* [sle-12-SP5-SAP-DVD-x86_64-create_hdd_sles4sap_gnome@64bit-2gbram](https://openqa.suse.de/t3644587)
* [sle-15-SP2-Online-x86_64-create_hdd_sles4sap_textmode@64bit-2gbram](https://openqa.suse.de/t3644588)
* [sle-15-SP2-Full-s390x-skip_registration@s390x-kvm-sle12](https://openqa.suse.de/t3644589)
* [sle-15-SP2-Full-s390x-skip_registration@s390x-zVM-vswitch-l2](https://openqa.suse.de/t3644590)
* [sle-15-Server-DVD-Updates-x86_64-qam-gnome@64bit](https://openqa.suse.de/t3644591)

**Comment by ldevulder** [comment]
@okurz thanks! This PR seems also to fix this issue: https://openqa.suse.de/tests/3641239#step/mr_test/101, retriggered with this PR: https://openqa.suse.de/tests/3645860

**Comment by nilxam** [comment]
@okurz https://progress.opensuse.org/issues/60467 looks forget to import `is_desktop_installed` from `version_utils`

File: lib/opensusebasetest.pm [hunk 1]
```diff
@@ -625,6 +626,237 @@ sub wait_grub_to_boot_on_local_disk {
...
+        select_console('x11');
+    }
+    else {
+        return $self->wait_boot_textmode(ready_time => $ready_time) if ($textmode || check_var('DESKTOP', 'textmode'));
+    }
+
+    # On SLES4SAP upgrade tests with desktop, only check for a DM screen with the SAP System
+    # Administrator user listed but do not attempt to login
+    if (get_var('HDDVERSION') and is_desktop_installed() and is_upgrade() and is_sles4sap()) {
+        assert_screen 'displaymanager-sapadm', $$ready_time;
```
Comment (ldevulder) [inline lib/opensusebasetest.pm]: I just saw this issue with `$$ready_time`... I'm fixing it!

**Pull request merged**